packages/sg-std/src/fees.rs

```diff
@@ -12,19 +12,17 @@ pub fn burn_and_distribute_fee(
     info: &MessageInfo,
     fee_amount: u128,
 ) -> Result<Vec<SubMsg>, FeeError> {
-    let payment = must_pay(info, NATIVE_DENOM)?;
+    let payment = may_pay(info, NATIVE_DENOM)?;
```

Comment:
Not sure why this switched because you only call this function if there is a fee...

Reply:
good catch, fixed